File: src/TaxAuthorityProvider/TaxAuthorities/Slovenia/ReportGeneration/DIV/CSV_Doh_DIV.py
from typing import Sequence

import pandas as pd

import TaxAuthorityProvider.TaxAuthorities.Slovenia.Schemas.Schemas as ss


def generateDataFrameReport(divLines: Sequence[ss.EDavkiDividendReportLine]) -> pd.DataFrame:
    def convertToDict(data: ss.EDavkiDividendReportLine):
        converted = {
            "Datum prejema dividend": data.DateReceived.format("YYYY-MM-DD"),
            "Davčna številka izplačevalca dividend": data.TaxNumberForDividendPayer,
            "Identifikacijska številka izplačevalca dividend": data.DividendPayerIdentificationNumber,
            "Naziv izplačevalca dividend": data.DividendPayerTitle,
            "Naslov izplačevalca dividend": data.DividendPayerAddress,
            "Država izplačevalca dividend": data.DividendPayerCountryOfOrigin,
            "Šifra vrste dividend": data.DividendType.value,
            "Znesek dividend (v EUR)": data.DividendAmount,
            "Znesek dividend (v Originalni Valuti)": data.DividendAmountInOriginalCurrency,
            "Tuji davek (v EUR)": data.ForeignTaxPaid,
            "Tuji davek (v Originalni Valuti)": data.ForeignTaxPaidInOriginalCurrency,
            "Država vira": data.CountryOfOrigin,
            "Uveljavljam oprostitev po mednarodni pogodbi": data.TaxReliefParagraphInInternationalTreaty,
            "Action Tracking": data.DividendIdentifierForTracking,
        }
        return converted

    dataAsDict = list(map(convertToDict, divLines))
    dataframe = pd.DataFrame.from_records(dataAsDict)
    return dataframe

File: lib/taxAuthorities/slovenia/reportGeneration/div/CsvDohDiv.ts
import type { EDavkiDividendReportLine } from "@brrr/taxAuthorities/slovenia/schemas/Schemas.ts";

type CsvRow = Record<string, unknown>;

export function generateCsvReport(divLines: EDavkiDividendReportLine[]): CsvRow[] {
	return divLines.map((data) => ({
		"Datum prejema dividend": data.dateReceived.toFormat("yyyy-MM-dd"),
		"Davčna številka izplačevalca dividend": data.taxNumberForDividendPayer,
		"Identifikacijska številka izplačevalca dividend": data.dividendPayerIdentificationNumber,
		"Naziv izplačevalca dividend": data.dividendPayerTitle,
		"Naslov izplačevalca dividend": data.dividendPayerAddress,
		"Država izplačevalca dividend": data.dividendPayerCountryOfOrigin,
		"Šifra vrste dividend": data.dividendType,
		"Znesek dividend (v EUR)": data.dividendAmount,
		"Znesek dividend (v Originalni Valuti)": data.dividendAmountInOriginalCurrency,
		"Tuji davek (v EUR)": data.foreignTaxPaid,
		"Tuji davek (v Originalni Valuti)": data.foreignTaxPaidInOriginalCurrency,
		"Država vira": data.countryOfOrigin,
		"Uveljavljam oprostitev po mednarodni pogodbi": data.taxReliefParagraphInInternationalTreaty,
		"Action Tracking": data.dividendIdentifierForTracking,
	}));
}
